### ISSUE TITLE
Write the tutorial

### ISSUE THREAD

**Issue by yaobinwen** [yaobinwen/cmake-explained#2]
# Description

This issue tracks my ideas of how to deliver the content in a more concise and clear way.

As of today (2023-05-21), I think a better way of organizing the topics is:
- Use a few sections to help the readers see the big picture, so when they have a question, they know where they are and can find the related topics.
- Help the readers know how to understand CMake. See `T1` below.
- CMake is big, so I should not try to cover everything in one sitting. Instead, I need to go iterations:
  - In the first iteration, I should just build a simple executable program without static or dynamic libraries. But this example includes not only the core building part, but also the related parts of the whole lifecycle, including:
    - Building Debug and Release.
    - Testing
    - Installing
    - Packaging and distributing
  - In the future iterations, I should gradually cover the more detailed topics, such as:
    - Build static/dynamic libraries.
    - Generator expressions
    - Export configuration.

- [ ] T1. Add chapter "Introduction: How to Understand CMake"
  - Software engineering is always a mixed effort.
  - Need to understand the build process.
  - Need to know what problems a build process may encounter.
  - Understand `target` and properties.
  - Understand various parts of CMake: it's a build tool, but also a programming language.
  - Know "modern CMake".
- [ ] T2. Move the current chapters about target and properties into the section "Targets and Properties: Center of the Stage".
- [ ] T3. Mention "Imported target" and "Alias target".
- [ ] T4: Iteration one: Build the core part:
  - [ ] T4.1: Write about the goal:
    - 1). Build a debug version of the code for testing.
    - 2). Build a release version of the code for distributing.
  - [ ] T4.2: Explain "targets" and "properties".
  - [ ] T4.3: Explain `Debug` and `Release`.
  - [ ] T4.4: Explain how to run tests and use the test dashboard.
  - [ ] T4.5: Explain how to packaging and distributing.
  - [ ] T4.6: Explain how to installing.

(To be continued.)

**Comment by yaobinwen** [comment]
# 01 - Introduction: How to Understand CMake

## TL;DR

> **Modern CMake** (which will be discussed further in later chapters), in contrast with the legacy CMake, tries to describe the build process using a **modular approach** in which `targets` are the building blocks (or "modules") and `properties` are the interface to fit the building blocks into the entire build process.

## Modulization: A Common Engineering Approach

(Talk about the benefits of modulization in engineering. See [All You Need to Know About Modularization](https://www.modularmanagement.com/blog/all-you-need-to-know-about-modularization).)

The point is: define standardized modules; standardize the interfaces so the modules can be integrated together.

## Legacy CMake vs Modern CMake

Legacy CMake tends to be "flat". See [my C/C++ code project](https://github.com/yaobinwen/robin_on_rails/tree/master/lang/c_cpp/code).

Modern CMake modulizes the modules: each module handles its own build process and then gets integrated into the entire build process.

## Understand the Build Process

Talk about the build process.

## How CMake Solves the Problems in the Build Process
